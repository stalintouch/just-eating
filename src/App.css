body, html{
  height: 100%;
  width: 100%;
  color: black;
  background: #F5F5F5;
  font-size:14px;
}

.container{
  width: 90%;
  display: block;
  margin: 0 auto;
}

.App-title{
  font-size: 2rem;
  text-align: center;
}

.App-title span{
  font-size: 3rem;
  font-family: cursive;
  border-radius: 10%;
}
a{
  color: rgb(0, 188, 212);
}

.card-ul{
  line-height: 200%;
  list-style: none;
  padding: 0;
}

.googlemaps{
  height: 300px;
  width: 40%;
  position: absolute;
  right: 1%;
  bottom: 2.5%;
}

.not-found{
  font-size: 30;
  color:red;
  text-align: center;
}