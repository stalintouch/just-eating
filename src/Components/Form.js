import React, { Component } from 'react';
import AutoComplete from 'material-ui/AutoComplete';

export default class Form extends Component {
  state = {
    input: '',
    cities: ["toronto"],
    searchText:''
  };

  handleInput = async e => {
    await this.setState({ input: e, searchText: e });
    const response = await fetch(
      `http://gd.geobytes.com/AutoCompleteCity?&filter=US,CA&q=${
        this.state.input
      }`
    );
    const data = await response.json();
    this.setState({ cities: data });
  };

  update = e => {
    this.props.updateCity(e);
  };

  render() {
    return (
        <AutoComplete
          autoFocus
          listStyle={{ maxHeight: 200, overflow: 'auto' }}
          floatingLabelText="Type a city and get restaurants"
          filter={AutoComplete.fuzzyFilter}
          dataSource={this.state.cities}
          maxSearchResults={5}
          onUpdateInput={this.handleInput}
          onNewRequest={this.update}
          disableFocusRipple={false}
          searchText={this.state.searchText}
          onChange={e=> console.log(e.target)}
        />
    );
  }
}
