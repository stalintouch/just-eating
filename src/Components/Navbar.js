import React, { Component } from 'react';
import AppBar from 'material-ui/AppBar';
import IconButton from 'material-ui/IconButton';
import FlatButton from 'material-ui/FlatButton';

export default class AppBarExampleIconButton extends Component {
  constructor(props) {
    super(props);
    this.handleClick = this.handleClick.bind(this);
  }
  // this function auto focus on the search bar when the navbar is clicked
  handleClick() {
    this.props.focus('autoComplete');
  }

  render() {
    const styles = {
      title: {
        cursor: 'pointer'
      }
    };

    return (
      <AppBar
        className="navbar"
        title={
          <span className="title-left" style={styles.title}>
            Just Eat it!
          </span>
        }
        onClick={this.handleClick}
        iconElementLeft={<IconButton />}
        iconElementRight={<FlatButton label="Find Restaurants nearby" />}
      />
    );
  }
}

