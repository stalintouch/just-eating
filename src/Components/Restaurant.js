import React, { Component } from 'react';
import fetchJSON from '../scripts/fetchJSON';
import RestaurantCard from './RestaurantCard';
import NotFound from './NotFound';
import CircularProgress from 'material-ui/CircularProgress';
import FlatButton from 'material-ui/FlatButton';
const imagesUrl =
  'https://api.unsplash.com/search/photos?client_id=cdaa537bca08095f8342feb8bc2e12893de05e62eba059daed8dba9e13e046bd&page=1&query=food&per_page=100';

class Restaurant extends Component {
  constructor(props) {
    super(props);
    this.state = { backgroundImages: [], loading: true };
    this.handleFocus = this.handleFocus.bind(this);
  }
  // fetches the background images from unsplash, to display them as background
  async componentDidMount() {
    const images = await fetchJSON(imagesUrl);
    console.log(images);
    this.setState({ backgroundImages: images.results });
  }

  componentWillReceiveProps() {
    // this settimeout is to make sure the loading spinner disappears after the component receive
    // props but not too soon so it shows the not found component
    setTimeout(() => {
      this.setState({ loading: false });
    }, 1000);
  }

  handleFocus() {
    this.props.focus('autoComplete');
  }
  // renderCards loops throught the props received from App.js and assembles a RestaurantCard with the
  // proper info passed
  renderCards = () => {
    return this.props.restaurants.map((restaurant, idx) => {
      return (
        <RestaurantCard
          backgroundImage={this.state.backgroundImages[idx].urls.full}
          key={restaurant.id}
          restaurantInfo={restaurant}
        />
      );
    });
  };

  render() {
    // shows the loading spinner
    if (!this.props.restaurants.length && this.state.loading) {
      return (
        <div style={{ display: ' flex', justifyContent: 'center' }}>
          <CircularProgress size={180} thickness={15} />
        </div>
      );
      // shows the not found component
    } else if (!this.props.restaurants.length && !this.state.loading) {
      return <NotFound />;
    } else {
      return (
        <div>
          <div style={{ marginBottom: '30px' }}>
            Not in {this.props.city}? Change it&nbsp;&nbsp;
            <FlatButton
              className="restaurant"
              label="Here"
              onClick={this.handleFocus}
              backgroundColor="#FFFF00"
            />
          </div>
          {/* shows the RestaurantCard component */}
          {this.renderCards()}
        </div>
      );
    }
  }
}

export default Restaurant;
