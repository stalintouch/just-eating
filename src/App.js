import React, { Component } from 'react';
import './App.css';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import Restaurant from './Components/Restaurant';
import fetchJSON from './scripts/fetchJSON';
import Form from './Components/Form';
import focus from './scripts/focus';
import Navbar from './Components/Navbar';
import data from './__test__/data/restaurantData.json';
const restaurantAPIUrl =
  'https://opentable.herokuapp.com/api/restaurants?city=';

class App extends Component {
  state = { city: '', restaurants: [] };

  async componentDidMount() {
    const locationUrl = 'https://ipinfo.io';
    const location = await fetchJSON(locationUrl, {
      accept: 'application/json',
      'Content-Type': 'application/json'
    });

    this.setState({ city: location.city || 'toronto' });
    // const restaurants = await fetchJSON(
    //   `${restaurantAPIUrl + this.state.city}`
    // );
    this.setState({ restaurants: data.restaurants });
  }

  updateInput = async inputCity => {
    var city = await inputCity
      .split(', ')
      .slice(0, 1)
      .join('');

    this.setState({ city });
    const restaurants = await fetchJSON(
      `${restaurantAPIUrl + this.state.city}`
    );
    this.setState({ restaurants: restaurants.restaurants });
  };

  render() {
    return (
      <MuiThemeProvider>
        <div className="container">
          <Navbar focus={focus} />
          <header className="App-header">
            <h1 className="App-title">
              Find the best restaurants, cafÃ©s, and bars in
              <span> {this.state.city}</span>
            </h1>
          </header>
          <Form updateCity={this.updateInput} />
          <Restaurant
            city={this.state.city}
            restaurants={this.state.restaurants}
            focus={focus}
          />
        </div>
      </MuiThemeProvider>
    );
  }
}

export default App;
