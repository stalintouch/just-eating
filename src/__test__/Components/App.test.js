import App from '../../App';
import Navbar from '../../Components/Navbar';
import Restaurant from '../../Components/Restaurant';
import Form from '../../Components/Form';

describe('<App />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App />);
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders the navbar component inside App correctly', () => {
    expect(wrapper.find(Navbar)).toHaveLength(1);
  });

  it('renders the Restaurant component inside App correctly', () => {
    expect(wrapper.find(Restaurant)).toHaveLength(1);
  });

  it('renders the Form component inside App correctly', () => {
    expect(wrapper.find(Form)).toHaveLength(1);
  });

  it('check state and calls componentDidMount  and checks the state changes between the two phases', async () => {
    // check if restaurants and city are empty on the state object
    expect(wrapper.state().restaurants).toEqual([]);
    expect(wrapper.state().city).toEqual('');

    // calls componentDidMount and update state
    await wrapper.instance().componentDidMount();

    // check if calling componentDidMount has updated both state by checking the length of the array and that city is not an empty string
    expect(wrapper.state().restaurants.length).toBeGreaterThan(1);
    expect(wrapper.state().city).toBeTruthy();
  });
});
