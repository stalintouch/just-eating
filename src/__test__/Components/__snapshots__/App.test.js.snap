// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div
        className="container"
>
        <AppBarExampleIconButton
                focus={[Function]}
        />
        <header
                className="App-header"
        >
                <h1
                        className="App-title"
                >
                        Find the best restaurants, cafés, and bars in
                        <span>
                                 
                                
                        </span>
                </h1>
        </header>
        <Form
                updateCity={[Function]}
        />
        <Restaurant
                city=""
                focus={[Function]}
                restaurants={Array []}
        />
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <AppBarExampleIconButton
            focus={[Function]}
/>,
          <header
            className="App-header"
>
            <h1
                        className="App-title"
            >
                        Find the best restaurants, cafés, and bars in
                        <span>
                                     
                                    
                        </span>
            </h1>
</header>,
          <Form
            updateCity={[Function]}
/>,
          <Restaurant
            city=""
            focus={[Function]}
            restaurants={Array []}
/>,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "focus": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1
              className="App-title"
>
              Find the best restaurants, cafés, and bars in
              <span>
                             
                            
              </span>
</h1>,
            "className": "App-header",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Find the best restaurants, cafés, and bars in",
                <span>
                   
                  
</span>,
              ],
              "className": "App-title",
            },
            "ref": null,
            "rendered": Array [
              "Find the best restaurants, cafés, and bars in",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    "",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  "",
                ],
                "type": "span",
              },
            ],
            "type": "h1",
          },
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "updateCity": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "city": "",
            "focus": [Function],
            "restaurants": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="container"
>
          <AppBarExampleIconButton
                    focus={[Function]}
          />
          <header
                    className="App-header"
          >
                    <h1
                              className="App-title"
                    >
                              Find the best restaurants, cafés, and bars in
                              <span>
                                         
                                        
                              </span>
                    </h1>
          </header>
          <Form
                    updateCity={[Function]}
          />
          <Restaurant
                    city=""
                    focus={[Function]}
                    restaurants={Array []}
          />
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <AppBarExampleIconButton
              focus={[Function]}
/>,
            <header
              className="App-header"
>
              <h1
                            className="App-title"
              >
                            Find the best restaurants, cafés, and bars in
                            <span>
                                           
                                          
                            </span>
              </h1>
</header>,
            <Form
              updateCity={[Function]}
/>,
            <Restaurant
              city=""
              focus={[Function]}
              restaurants={Array []}
/>,
          ],
          "className": "container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "focus": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h1
                className="App-title"
>
                Find the best restaurants, cafés, and bars in
                <span>
                                 
                                
                </span>
</h1>,
              "className": "App-header",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Find the best restaurants, cafés, and bars in",
                  <span>
                     
                    
</span>,
                ],
                "className": "App-title",
              },
              "ref": null,
              "rendered": Array [
                "Find the best restaurants, cafés, and bars in",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      " ",
                      "",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    " ",
                    "",
                  ],
                  "type": "span",
                },
              ],
              "type": "h1",
            },
            "type": "header",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "updateCity": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "city": "",
              "focus": [Function],
              "restaurants": Array [],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
